RFS/src/main/java/com/rfs/worker/IndexRunner.java

```diff
@@ -19,18 +20,28 @@ public class IndexRunner {
     private final IndexMetadata.Factory metadataFactory;
     private final IndexCreator_OS_2_11 indexCreator;
     private final Transformer transformer;
+    private final List<String> indexAllowlist;
 
     public void migrateIndices() {
         SnapshotRepo.Provider repoDataProvider = metadataFactory.getRepoDataProvider();
         // TODO - parallelize this, maybe ~400-1K requests per thread and do it asynchronously
-        for (SnapshotRepo.Index index : repoDataProvider.getIndicesInSnapshot(snapshotName)) {
-            var indexMetadata = metadataFactory.fromRepo(snapshotName, index.getName());
-            var root = indexMetadata.toObjectNode();
-            var transformedRoot = transformer.transformIndexMetadata(root);
-            var resultOp = indexCreator.create(transformedRoot, index.getName(), indexMetadata.getId());
-            resultOp.ifPresentOrElse(value -> log.info("Index " + index.getName() + " created successfully"),
-                    () -> log.info("Index " + index.getName() + " already existed; no work required")
-            );
-        }
+        
+        BiConsumer<String, Boolean> logger = (indexName, accepted) -> {
+            if (!accepted) {
+                log.info("Index " + indexName + " rejected by allowlist");
+            }
+        };
+        repoDataProvider.getIndicesInSnapshot(snapshotName).stream()
+            .filter(FilterScheme.filterIndicesByAllowList(indexAllowlist, logger))
+            .peek(index -> {
+                var indexMetadata = metadataFactory.fromRepo(snapshotName, index.getName());
+                var root = indexMetadata.toObjectNode();
+                var transformedRoot = transformer.transformIndexMetadata(root);
+                var resultOp = indexCreator.create(transformedRoot, index.getName(), indexMetadata.getId());
+                resultOp.ifPresentOrElse(value -> log.info("Index " + index.getName() + " created successfully"),
+                        () -> log.info("Index " + index.getName() + " already existed; no work required")
+                );
+            })
+            .count(); // Force the stream to execute
```

Comment:
foreach() will force the stream to run.  I think that this will generate a linting error as it is since the output of count() is discarded.